### ISSUE TITLE
Azure - AppServiceCertificateBinding fails with Thumbprint error

### ISSUE THREAD

**Issue by luke-lewandowski** [hashicorp/terraform-cdk#1086]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

"cdktf": "^0.6.3",
"cdktf-cli": "^0.6.3",
"constructs": "^10.0.5"
"azurerm@~> 2.77.0"

### Affected Resource(s)

**AppServiceCertificateBinding**

### Debug Output

[https://gist.github.com/luke-lewandowski/f76f813b64949d4aba0cf6436aa9590d](https://gist.github.com/luke-lewandowski/f76f813b64949d4aba0cf6436aa9590d)

### Expected Behavior
I'm creating a shared AppServicePlan, then creating AppService with a custom domain name which then gets bound together with a managed SSL certificate. It appears that the 
On the same version yesterday this same code was working and I'm not sure what has changed. 

### Actual Behavior

The build fails with a thumbprint being missing. Although - I'm not able to specify that record in the binding.

### Steps to Reproduce

I'm attaching a snippet that creates relevant resources based on the existing appServicePlan and resource group.

[https://gist.github.com/luke-lewandowski/a031761a89324c79cb2aed19283d3987](https://gist.github.com/luke-lewandowski/a031761a89324c79cb2aed19283d3987)

### Important Factoids

N/A

### References

- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_certificate_binding

- #0000

**Comment by jsteinich** [comment]
It looks like your `AppServiceCustomHostnameBinding` isn't depending on the `DnsTxtRecord` correctly. If you assign the `DnsTxtRecord` to a variable, you can pass it to the `dependsOn` array when constructing the `AppServiceCustomHostnameBinding`.

**Comment by luke-lewandowski** [comment]
@jsteinich - Thanks for your prompt response. I have updated the template as per what you have suggested.

The error is not with CustomHostnameBinding. It is about the Certificate Binding.


```
[2021-09-29T17:02:53.744] [ERROR] default - ╷
│ Error: creating/updating Custom Hostname Certificate Binding "api.dev.givvable.me" with certificate name "api.dev.givvable.me" (App Service "dev-test-api-as" / Resource Group "dev-test-rg"): web.AppsClient#CreateOrUpdateHostNameBinding: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="Parameter Thumbprint is null or empty." Details=[{"Message":"Parameter Thumbprint is null or empty."},{"Code":"BadRequest"},{"ErrorEntity":{"Code":"BadRequest","ExtendedCode":"51011","Message":"Parameter Thumbprint is null or empty.","MessageTemplate":"Parameter {0} is null or empty.","Parameters":["Thumbprint"]}}]
│ 
│   with azurerm_app_service_certificate_binding.supplier-certiicate-binding,
│   on cdk.tf.json line 176, in resource.azurerm_app_service_certificate_binding.supplier-certiicate-binding:
│  176:       }

```

**Comment by luke-lewandowski** [comment]
Updated code
```typescript
    const baseConfig = stackConfig.baseConfig;

    const rg = new DataAzurermResourceGroup(
      this,
      baseConfig.resourceGroupName,
      { name: baseConfig.resourceGroupName }
    );
    const ap = new DataAzurermAppServicePlan(
      this,
      baseConfig.appServicePlanName,
      { name: baseConfig.appServicePlanName, resourceGroupName: rg.name }
    );

    const asName = this.createResourceName("as");
    const as = new AppService(this, asName, {
      name: asName,
      location: rg.location,
      resourceGroupName: rg.name,
      appServicePlanId: ap.id,
      httpsOnly: true,
      clientAffinityEnabled: true,
      siteConfig: [
        {
          scmType: "VSTSRM",
          alwaysOn: true,
          linuxFxVersion: `DOCKER|${stackConfig.dockerImage}`,
          // healthCheckPath: '/health',
        },
      ],
      appSettings: locals.env_variables,
    });

    const dnsZoneName = this.createResourceName(`dnszone`);
    const dnsZone = new DataAzurermDnsZone(this, dnsZoneName, {
      name: stackConfig.domainZone,
      resourceGroupName: "givvable-core",
    });

    const dnsCnameRecordName = this.createResourceName(`cname`);
    const dnsCnameRecord = new DnsCnameRecord(this, dnsCnameRecordName, {
      name: stackConfig.subdomain,
      zoneName: dnsZone.name,
      resourceGroupName: dnsZone.resourceGroupName,
      ttl: 300,
      record: as.defaultSiteHostname,
    });

    const dnsTxtRecordName = this.createResourceName("txt");
    const dnsTxtRecord = new DnsTxtRecord(this, dnsTxtRecordName, {
      name: `asuid.${dnsCnameRecord.name}`,
      zoneName: dnsZone.name,
      resourceGroupName: dnsZone.resourceGroupName,
      ttl: 300,
      record: [
        {
          value: as.customDomainVerificationId,
        },
      ],
    });

    const customBindingName = this.createResourceName(
      "custom-hostname-binding"
    );
    const customAsHostNameBinding = new AppServiceCustomHostnameBinding(
      this,
      customBindingName,
      {
        hostname: `${stackConfig.subdomain}.${stackConfig.domainZone}`,
        appServiceName: as.name,
        resourceGroupName: rg.name,
        lifecycle: {
          ignoreChanges: ["ssl_state", "thumbprint"],
        },
        dependsOn: [dnsCnameRecord, dnsTxtRecord],
      }
    );

    const asmcName = this.createResourceName("cert-binding");
    const appCertificate = new AppServiceManagedCertificate(this, asmcName, {
      customHostnameBindingId: customAsHostNameBinding.id,
    });

    new AppServiceCertificateBinding(this, "supplier-certiicate-binding", {
      hostnameBindingId: customAsHostNameBinding.id,
      certificateId: appCertificate.id,
      sslState: "SniEnabled",
      dependsOn: [appCertificate, dnsCnameRecord],
    });
```

**Comment by luke-lewandowski** [comment]
This issue has been resolved by upgrading the provider to 2.78.0. Thanks!

**Comment by ansgarm** [comment]
Perfect 👍
